### PR TITLE
Adjust Pairing Adapter to create new deployments per adapter configuration

### PR DESCRIPTION
**Pull request by NickyMateev**
# Overview

**Description**

This PR proposes a solution for creation of new Pairing Adapter deployments for each new adapter configuration that's introduced.

### CODE REVIEW

File: chart/compass/charts/pairing-adapter/templates/virtual-service.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: {{ template "fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    release: {{ .Release.Name }}
```
Comment (NickyMateev) [inline chart/compass/charts/pairing-adapter/templates/virtual-service.yaml:7]: Should we add additional labels like `helm` or k8s labels here or something? https://github.com/kyma-incubator/compass/blob/ac249a4671d2099d7c172ce130e749bc281b7a6f/chart/compass/charts/connector/templates/virtual-service.yaml#L8

File: chart/compass/charts/pairing-adapter/templates/deployment.yaml [hunk 2]
```diff
@@ -28,30 +32,30 @@ spec:
...
-                value: {{ .Values.deployment.envs.mappingTemplateJSONBody | quote }}
+                value: {{ $config.envs.mappingTemplateJSONBody | quote }}
               - name: MAPPING_TEMPLATE_TOKEN_FROM_RESPONSE
-                value: {{ .Values.deployment.envs.mappingTemplateTokenFromResponse | quote }}
+                value: {{ $config.envs.mappingTemplateTokenFromResponse | quote }}
               - name: OAUTH_URL
                 valueFrom:
                     secretKeyRef:
-                        name: {{.Values.deployment.oauthSecret}}
+                        name: {{ $config.oauthSecret}}
```
Comment (PetarTodorovv) [inline chart/compass/charts/pairing-adapter/templates/deployment.yaml:45]: Missing space

File: chart/compass/charts/pairing-adapter/values.yaml [hunk 3]
```diff
@@ -1,18 +1,21 @@
...
-  envs:
-    mappingTemplateExternalURL: ""
-    mappingTemplateHeaders: ""
-    mappingTemplateJSONBody: ""
-    mappingTemplateTokenFromResponse: ""
-    oauthStyle: "AuthDetect"
   strategy: {} # Read more: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy
   nodeSelector: {}
+  adapters:
+    adapter0:
```
Comment (desislavaa) [inline chart/compass/charts/pairing-adapter/values.yaml:13]: `example` also seems applicable to me.
Maybe rename the tenant fetcher job as well (it seems like it uses the same declaration principle) - either to `example`, or `job0`

**Review by desislavaa — DISMISSED** [review]

**Review by desislavaa — APPROVED** [review]

**Pull request merged**